### PR TITLE
Fuzz test for GEOPM hash function

### LINKED ISSUE #3003: Figure out interfaces that we will fuzz test first, enable first fuzz testing harness
> estimate 2

Done criteria:
- Fuzz testing harness for first API developed and in place
- Fuzz test should be incorporated into build via a configure option (to be built manually/explicitly, not as part of CI)
- Look into where we could deploy fuzz tests, understand requirements and scope of work to deploy
- Run fuzz test, confirm it works

### PR DESCRIPTION
**Pull request by avilcheslopez**
- Fixes #3003

### CODE REVIEW

File: service/Makefile.am [hunk 1]
```diff
@@ -122,6 +127,12 @@ geopmwrite_SOURCES = src/geopmwrite_main.cpp
 geopmwrite_CXXFLAGS = $(AM_CXXFLAGS) -std=c++17
 geopmwrite_LDADD = libgeopmd.la
 
+if ENABLE_FUZZTESTS
```
Comment (cmcantalupo) [inline service/Makefile.am]: Maybe better to put this change in ``service/integration/Makefile.mk`` and move ``service/src/fuzztest_geopmhash_main.cpp`` to ``service/integration/test/fuzztest_geopmhash_main.cpp``?

Comment (avilcheslopez) [reply service/Makefile.am]: Will do

**Review by cmcantalupo — DISMISSED** [review]

File: service/integration/test/fuzztest_geopmhash_main.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <cstdint>
+
```
Comment (bgeltz) [inline service/integration/test/fuzztest_geopmhash_main.cpp]:
```suggestion
```

**Review by bgeltz — COMMENTED** [review]
Tiny nit.  LGTM otherwise

**Comment by bgeltz** [comment]
What's the expectation here if someone is building with the GNU toolchain but configures with ``--enable-fuzz``?
Also, what about configuring with ``--disable-debug``?

I think this is the first time we're adding support for CLANG specific features, and we may want to take care to isolate them properly.

**Review by bgeltz — APPROVED** [review]

**Pull request merged**